contracts/key_store/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+#![cfg_attr(not(any(feature = "std", test)), no_std)]
+
+extern crate alloc;
+
+pub use crate::key_store::{KeyStore, KeyStoreRef};
+
+#[ink::contract(env = pink_extension::PinkEnvironment)]
+mod key_store {
+    use alloc::{vec, vec::Vec};
+    use scale::{Decode, Encode};
+
+    #[derive(Encode, Decode, Debug, PartialEq, Eq)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        BadOrigin,
+        MissingExecutor,
+    }
+
+    type Result<T> = core::result::Result<T, Error>;
+
+    #[ink(storage)]
+    pub struct KeyStore {
+        pub admin: AccountId,
+        pub prv_keys: Vec<[u8; 32]>,
+        pub executor: Option<AccountId>,
+    }
+
+    impl Default for KeyStore {
+        fn default() -> Self {
+            Self::default()
+        }
+    }
+
+    impl KeyStore {
+        #[allow(clippy::should_implement_trait)]
+        #[ink(constructor)]
+        pub fn default() -> Self {
+            let mut prv_keys: Vec<[u8; 32]> = vec![];
+
+            for index in 0..5 {
+                let private_key = pink_web3::keys::pink::KeyPair::derive_keypair(
+                    &[b"worker".to_vec(), [index].to_vec()].concat(),
+                )
+                .private_key();
+                prv_keys.push(private_key);
+            }
+
+            Self {
+                admin: Self::env().caller(),
+                prv_keys,
+                executor: None,
+            }
+        }
+
+        #[ink(message)]
+        pub fn set_executor(&mut self, executor: AccountId) -> Result<()> {
+            self.ensure_owner()?;
+            self.executor = Some(executor);
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn remove_executor(&mut self) -> Result<()> {
+            self.ensure_owner()?;
+            self.executor = None;
+            Ok(())
+        }
+
+        /// Only whitelisted executors are allowed to call this function
```

Comment:
but the comment here states that the executors are plural

Reply:
Updated